pkg/apis/resolution/v1beta1/resolution_request_types.go

```diff
@@ -62,6 +62,7 @@ type ResolutionRequestSpec struct {
 	// resource being requested. For example: repo URL, commit SHA,
 	// path to file, the kind of authentication to leverage, etc.
 	// +optional
+	// +listType=atomic
```

Comment:
Hey, this was me too! See #4402 - dealing with API rule violations from `openapi-gen`. I went with `atomic` because that seemed like the sanest default choice.